libs/vgc/workspace/workspace.h

```diff
@@ -0,0 +1,161 @@
+// Copyright 2022 The VGC Developers
+// See the COPYRIGHT file at the top-level directory of this distribution
+// and at https://github.com/vgc/vgc/blob/master/COPYRIGHT
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef VGC_WORKSPACE_WORKSPACE_H
+#define VGC_WORKSPACE_WORKSPACE_H
+
+#include <map>
+#include <unordered_map>
+
+#include <vgc/core/animtime.h>
+#include <vgc/core/flags.h>
+#include <vgc/core/id.h>
+#include <vgc/core/object.h>
+#include <vgc/dom/document.h>
+#include <vgc/graphics/engine.h>
+#include <vgc/topology/vac.h>
+#include <vgc/workspace/api.h>
+#include <vgc/workspace/element.h>
+
+namespace vgc::workspace {
+
+VGC_DECLARE_OBJECT(Workspace);
+
+namespace detail {
+
+struct VacElementLists;
+
+} // namespace detail
+
+/// \class vgc::workspace::Workspace
+/// \brief Represents a workspace to manage, manipulate and visit a vector graphics scene.
+///
+class VGC_WORKSPACE_API Workspace : public core::Object {
```

Comment:
Add comment: "This implementation is still preliminary and the API may change significantly"